ivy/functional/ivy/data_type.py

```diff
@@ -692,19 +692,99 @@ def closest_valid_dtype(type: Union[ivy.Dtype, str, None]) -> Union[ivy.Dtype, s
 def is_int_dtype(
     dtype_in: Union[ivy.Dtype, str, ivy.Array, ivy.NativeArray, Number]
 ) -> bool:
-    """Determine whether the input data type is an int data-type.
+    """
+    Determine whether the input data type is an int data type.
 
     Parameters
     ----------
     dtype_in
-        Datatype to test
+        input data type to test.
 
     Returns
     -------
     ret
-        Whether or not the data type is an integer data type
+        "True" if the input data type is an integer, otherwise "False".
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this function is *nestable*, and therefore also accepts :code:`ivy.Container`
+    instances in place of any of the arguments.
+
+    Examples
+    --------
+     With :code:`ivy.Dtype` input:
+
+    >>> x = ivy.is_int_dtype(ivy.int8)
+    >>> print(x)
+    True
+
+    >>> x = ivy.is_int_dtype(ivy.int32)
+    >>> print(x)
+    True
+
+    >>> x = ivy.is_int_dtype(ivy.uint64)
+    >>> print(x)
+    True
+
+    >>> x = ivy.is_int_dtype(ivy.float64)
+    >>> print(x)
+    True
+
+    >>> x = ivy.is_int_dtype(ivy.bool)
+    >>> print(x)
+    False
+
+    With :code:`str` input:
+
+    >>> x = "1"
+    >>> print(ivy.is_int_dtype(x))
+    True
+
+    >>> x = "int"
+    >>> print(ivy.is_int_dtype(x))
+    False
+
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([1., 2., 3.])
+    >>> print(x.dtype)
+    float32
+
+    >>> print(ivy.is_int_dtype(x))
+    False
+
+    With :code:`ivy.NativeArray` input:
+
+    >>> x = ivy.native_array([[-1, -1, -1], [1, 1, 1]], \
+        dtype='int16')
```

Comment:
`dtype = ivy.int16` and not 'int16'

Reply:
I changed this but still it is failing :(